engine/runtime-parser/src/main/java/org/enso/compiler/core/Persistance.java

```diff
@@ -0,0 +1,534 @@
+package org.enso.compiler.core;
+
+import java.io.ByteArrayOutputStream;
+import java.io.DataInput;
+import java.io.DataInputStream;
+import java.io.DataOutput;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.ByteBuffer;
+import java.util.HashMap;
+import java.util.IdentityHashMap;
+import java.util.Map;
+import java.util.ServiceLoader;
+
+import org.enso.compiler.core.Persistance.Reference;
+
+public abstract class Persistance<T> {
```

Comment:
Missing some documentation about individual subclasses and their methods on what they are doing and e.g. how `writeXYZ` differ and when they should be used.

Reply:
Done in https://github.com/enso-org/enso/pull/8207#discussion_r1398105136